.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,170 @@
+name: build-and-test
+
+on:
+  push:
+    branches: [ "*" ]
+    tags: [ "v*" ]
+  pull_request:
+  workflow_dispatch:
+
+concurrency:
+  group: '${{ github.workflow }}-${{ github.job }}-${{ github.head_ref || github.ref_name }}'
+  cancel-in-progress: true
+
+env:
+  BUNDLER_VER: 2.6.1
+  RUBYGEMS_VER: 3.6.1
+
+jobs:
+  validate-no-compilation-required:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - windows-2022
+          - macos-14
+          - ubuntu-20.04
+          - ubuntu-22.04
+        ruby:
+          - 3.1.2
+          - 3.3
+        include:
+          - os: windows-2022
+            platform: x64-mingw32
+          - os: windows-2022
+            platform: x64-mingw-ucrt
+          - os: macos-14
+            platform: x86_64-darwin
+          - os: macos-14
+            platform: arm64-darwin
+          - os: ubuntu-22.04
+            platform: x86_64-linux
+          - os: ubuntu-22.04
+            platform: any
+          - os: ubuntu-20.04
+            platform: x86_64-linux
+          - os: ubuntu-20.04
+            platform: any
+            
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
+
+      - run: gem update --system $RUBYGEMS_VER
+
+      - name: Install gem dependencies
+        env:
+          CC: false
+          LD: false
+          CXX: false
+        run: bundle install
+
+      - name: Run tests
+        run: bundle exec rake rspec
+
+  # since we are packaging a pure ruby gem, we only need to package
+  # once for each ruby version, not once on every platform.
+  #
+  # and maybe we don't even need to package for each ruby version?
```

Comment:
As long as the same fig code works on all supported versions of Ruby, which the tests prove, then we should only need to build one gem.